File: src/dataset/c/armstrong-numbers_exercism-fiplox/openAI/ChatGPT4_armstrong_numbers_v3.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int copy = candidate, sum = 0, digits = 0, digit_powers[10];

    while (copy > 0) {
        digits++;
        copy /= 10;
    }

    for (int d = 0; d < 10; d++) {
        digit_powers[d] = pow(d, digits);
    }

    copy = candidate;
    while (copy > 0) {
        sum += digit_powers[copy % 10];
        copy /= 10;
    }

    return sum == candidate;
}

File: src/dataset/Java/armstrong-numbers_exercism-awsaf174/openAI/ChatGPT4_ArmstrongNumbers_v3.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int temp = numberToCheck, sum = 0, digits = (int) Math.log10(numberToCheck) + 1;
        while (temp != 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, digits);
            temp /= 10;
        }
        return sum == numberToCheck;
    }

}